src/main/java/jenkins/plugins/slack/SlackNotifier.java

```diff
@@ -184,8 +240,9 @@ public FormValidation doTestConnection(@QueryParameter("slackTeamDomain") final
                 @QueryParameter("slackRoom") final String room,
                 @QueryParameter("slackBuildServerUrl") final String buildServerUrl) throws FormException {
             try {
-                SlackService testSlackService = new StandardSlackService(teamDomain, authToken, room);
-                String message = "Slack/Jenkins plugin: you're all set on " + buildServerUrl;
+                SlackService testSlackService = new StandardSlackService(StringUtils.isEmpty(teamDomain) ? this.teamDomain : teamDomain,
```

Comment:
I understand what's going on here but it looks kind of obfuscated to an untrained eye.  Is there a better way you can set up checking for the string and falling back to a default?  Perhaps conditionals before the `testSlackService` variable would be a better.